cidc_api/resources/permissions.py

```diff
@@ -28,7 +28,7 @@
 
 @permissions_bp.route("/", methods=["GET"])
 @requires_auth("permissions")
-@use_args({"user_id": fields.Str()}, location="query")
+@use_args({"user_id": fields.Int()}, location="query")
```

Comment:
See line 50 below - the code was comparing a string to an integer, so `current_user.id != user_id` was always `True`.